third_party/proton/csrc/lib/Profiler/Instrumentation/InstrumentationProfiler.cpp

```diff
@@ -254,7 +254,8 @@ void InstrumentationProfiler::exitInstrumentedOp(uint64_t streamId,
                         normalizedDuration, kernelId, functionName,
                         blockTrace.blockId, blockTrace.procId, trace.uid,
                         device, static_cast<uint64_t>(runtime->getDeviceType()),
-                        timeShiftCost));
+                        timeShiftCost, blockTrace.initialTime,
+                        blockTrace.preFinalTime, blockTrace.postFinalTime));
```

Comment:
I feel somewhat unnatural when writing this code as these three values are essentially the same across the block, but we still need to repeat it and squeeze it into `class CycleMetric`, just to take it out later to a different `struct BlockTrace` instance. 

The dataflow looks like the following: 
```
           decode<I64Entry>                addMetric           getInt64Value
raw bytes ==================> BlockTrace ============> Metric ==============> BlockTrace
```

Not sure if there is a better way to do this. Or maybe it's just the cost of abstraction with the `addMetric` interface.

Reply:
> I feel somewhat unnatural when writing this code as these three values are essentially the same across the block

Well, even `blockId` is the same for all events within the same CTA. IMO it's fine unless we define new metrics